:memo: Rework apex domains (#2762)
platformsh/platformsh-docs: docs/src/domains/steps/dns.md

 
In such situations, using `A` records can cause downtime.
`A` records map a domain to an IP address.
When a router's IP address changes,
you need to manually update your `A` records.
Your site remains offline until you do,
or until the router is back from maintenance.
 
To prevent downtime, create `CNAME` records through your DNS provider.
`CNAME` records map a domain to another.
Set up your `CNAME` records to point at your project's [target URL](../../domains/steps/_index.md#2-get-the-target-for-your-project).
The DNS record for this target URL automatically resolves to the IP addresses of the routers in your project's region.
When a router's IP address changes,
the DNS record is automatically and immediately updated.
This prevents downtime without intervention from you.
 
Note that `CNAME` records can't point to [apex domains](../../other/glossary.md#apex-domain).
Check with your DNS provider if they offer [workarounds](#workarounds-for-apex-domains).
 
## Workarounds for apex domains

Although `CNAME` records can't point to apex domains,
most DNS providers offer workarounds.
Contact your DNS provider and choose the option that works for you.

> Note that workarounds are also available for use with a CDN.
> For example, [Fastly](../cdn/fastly.md) offers [Anycast options](https://docs.fastly.com/en/guides/using-fastly-with-apex-domains) 
> and [Cloudflare](../cdn/cloudflare.md) offers [`CNAME` flattening options](https://developers.cloudflare.com/dns/additional-options/cname-flattening/).

Reviewer: :hammer_and_wrench: :arrow_forward: With #2746, I'd be in favor of keeping the exact explanation of what the workaround is in one place. I know it's unlikely they'll change the name or anything, so maybe I shouldn't worry so much about repeating content. But I'd probably go with something like this:

```suggestion
If you use a CDN, you likely already have a workaround in place,
such as with [Fastly](../cdn/fastly.md#3-handle-apex-domains)
and [Cloudflare](../cdn/cloudflare.md#3-handle-apex-domains).
```

Not a blocking idea though.

---
title: "DNS management"
weight: 1
description: See why `CNAME` records are used and how to handle apex domains
sidebarTitle: "DNS management"
---
 
Using `CNAME` records helps [prevent downtime](#cname-records) during maintenance operations.
But as `CNAME` records can't point to apex domains,
you need to [set up a workaround](#workarounds-for-apex-domains).
Available workarounds depend on your DNS provider.

## `CNAME` records

Each site on Platform.sh is made up of a set of containers.
To map incoming requests to the appropriate container,
Platform.sh runs routers in [each region](../../development/regions.md).
A router's IP address can change in two cases:
- During an upgrade or maintenance operation, routers can be taken offline while changes are applied.
- During a region upscale or downscale, routers can be added or removed.

To prevent downtime in such situations,
create `CNAME` records through your DNS provider.
 
`CNAME` records map a domain to another.
Set up your `CNAME` records to point at your project's [target edge hostname](../../domains/steps/_index.md#2-get-the-target-for-your-project).
Each Platform.sh edge hostname comes with a DNS record

> that resolves the IP addresses of the routers in your project's region.

Author: Don't really like this sentence, but couldn't find a way to make it clearer so definitely open to suggestions.

If you want your site to be accessible at a URL like `https://example.com` and not only `https://www.example.com`,
this is the best way to do so.

To configure your domain name to point to your project using custom records, follow the instructions on [how to set up a custom domain](./_index.md).
When you come to configuring your DNS provider, replace the suggested `CNAME` record with the custom record pointing from your domain to the target.

Examples of such workaround records and providers include:

<!-- vale Platform.condescending = NO -->
* `CNAME` flattening at [CloudFlare](https://developers.cloudflare.com/dns/additional-options/cname-flattening)
* `ANAME` records at [easyDNS](https://easydns.com/features/aname-root-domain-alias/),
  [DNS Made Easy](https://support.dnsmadeeasy.com/support/solutions/articles/47001001412-aname-records),
  and [Name.com](https://www.name.com/support/articles/115010493967-adding-an-aname-alias-record)
* `ALIAS` records at [DNSimple](https://support.dnsimple.com/articles/alias-record/)
  and [ClouDNS](https://www.cloudns.net/wiki/article/18/)
<!-- vale Platform.condescending = YES -->

> 

Author: I went back and forth on whether I should keep all the links or not. Open to opinions.

>    `http://example.com` redirects to `http://www.example.com`, not to `https://www.example.com`.

Reviewer: :question: Is it necessary to have `http` redirects for the redirection service? Because generally, `http` is automatically redirected to `https` and so `http` doesn't need to be included in the routes definition. If it does for redirects, that should be a separate step. If it's not necessary, I wouldn't use the example of `http`, since many people won't be defining that.

To use `A` records, follow these steps:
 
1.  To get the IP addresses of your project's production environment, run the following command:

>     ```bash

Reviewer: :clipboard: 
````suggestion

        ```bash
````

Author: I applied this suggestion but ESLint then showed an error in my file (could be that my settings are off again? 🤔 ). How many spaces are we truly expecting here?
